### DISCUSSION TITLE
Remove spinner once LSP is completed

### DISCUSSION THREAD

**Discussion by ryanmsnyder** [linrongbin16/lsp-progress.nvim#59] [Q&A] [answered]
Thanks for the plugin - really like it so far.

Is it possible to remove the spinner once an lsp process is complete? I have my `decay` set to `3000`, which is quite a bit longer than the default. However, even though a process is complete, the spinner is still active. I'd like achieve a format where the spinner is replaced by a check mark once the process is complete, similar to how fidget.nvim works.

**Comment by linrongbin16** [comment, marked as answer]
emmmm, I think it's possible, just a little bit complicated to configure.

You will need 2 steps:

1. In [series_format](https://github.com/linrongbin16/lsp-progress.nvim/blob/main/lua/lsp-progress/defaults.lua#L78), you should return an object.

   ```lua
   series_format = function(title, message, percentage, done)

     -- same with default
     ...

     return {msg=table.concat(builder, " "), done=done}
   end
   ```
   
   This is for returning an extra value `done`, we will use it in second step.

2. In [client_format](https://github.com/linrongbin16/lsp-progress.nvim/blob/main/lua/lsp-progress/defaults.lua#L113), you can use below code:

   ```lua
   client_format = function(client_name, spinner, series_messages)
     if #series_messages == 0 then
       return nil
     end
     local builder = {}
     for _, series in ipairs(series_messages) do
       if series.done then
         spinner = '✓' -- replace your check mark
       end
       table.insert(builder, series.msg)
     end
     return "[" .. client_name .. "] " .. spinner .. " " .. table.concat(
       builder,
       ", "
     )
    end,
   ```


Note: I write this without any testing, please try to make it correct.

**Reply by linrongbin16** [reply]
in current design, the spinng icon is on client level. e.g. a lsp client only have 1 spining icon.

when you have multiple series messages, you need to decide whether to show the check mark:
1. show check mark when all the series messages are done.
2. show check mark when any of the series message is done.

it's your choice.

**Reply by ryanmsnyder** [reply]
@linrongbin16 thank you! This was very helpful

**Reply by linrongbin16** [reply]
could you share your final configs? I can add your use case to the advanced configs.
